lib/bolt_server/transport_app.rb

```diff
@@ -73,6 +73,38 @@ def run_command(target, body)
       @executor.run_command(target, command)
     end
 
+    def upload_file(target, body)
+      error = validate_schema(@schemas["action-upload_file"], body)
+      return [400, error.to_json] unless error.nil?
+
+      files = body['files']
+      destination = body['destination']
+      job_id = body['job_id']
+      cache_dir = @file_cache.create_cache_dir(job_id.to_s)
+      FileUtils.mkdir_p(cache_dir)
+      files.each do |file|
+        relative_path = file['relative_path']
+        uri = file['uri']
+        sha256 = file['sha256']
+        kind = file['kind']
+        path = File.join(cache_dir, relative_path)
+        if kind == 'file'
+          # The parent should already be created by `directory` entries,
+          # but this is to be on the safe side.
+          parent = File.dirname(path)
+          FileUtils.mkdir_p(parent)
+          @file_cache.download_file(path, sha256, uri)
+        elsif kind == 'directory'
+          # Create directory in cache so we can move files in.
+          FileUtils.mkdir_p(path)
+        else
```

Comment:
does this need to worry about the symlink support that @McdonaldSeanp just added?

Reply:
Per @McdonaldSeanp, the links will appear as normal files and directories here, so I'm under the assumption that link support is not needed.